src/components/Card.js

```diff
@@ -18,28 +28,45 @@ export default class Card {
     this._cardTitle.textContent = this._name;
     this._cardImage.alt = this._name;
     this._cardImage.src = this._link;
-
     this._setEventListeners();
+    this._setInitialLikeStatus();
 
     return this._cardElement;
   }
 
+  _setInitialLikeStatus() {
+    if (this._isLiked) {
+      this._cardLikeIcon.classList.add("card__like-button_pressed");
+    }
+  }
+
+  _updateLikeStatus = () => {
+    if (this._isLiked) {
+      this._cardLikeIcon.classList.remove("card__like-button_pressed");
+      this._handleCardLikeClick({
+        cardId: this._id,
+        method: "DELETE",
+        cardEl: this._cardElement,
+      });
+      this._isLiked = false;
+    } else {
+      this._cardLikeIcon.classList.add("card__like-button_pressed");
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.